lightning/src/ln/channel.rs

```diff
@@ -4788,6 +4788,39 @@ impl<Signer: Sign> Channel<Signer> {
 		self.channel_state >= ChannelState::FundingSent as u32
 	}
 
+	/// Returns true if the channel is awaiting the persistence of the initial ChannelMonitor.
+	/// If the channel is outbound, this implies we have not yet broadcasted the funding
+	/// transaction. If the channel is inbound, this implies simply that the channel has not
+	/// advanced state.
+	pub fn is_awaiting_initial_mon_persist(&self) -> bool {
+		if !self.is_awaiting_monitor_update() { return false; }
+		if self.channel_state &
+			!(ChannelState::TheirChannelReady as u32 | ChannelState::PeerDisconnected as u32 | ChannelState::MonitorUpdateFailed as u32)
+				== ChannelState::FundingSent as u32 {
+			// If we're not a 0conf channel, we'll be waiting on a monitor update with only
+			// FundingSent set, though our peer could have sent their channel_ready.
+			debug_assert!(self.minimum_depth.unwrap_or(1) > 0);
+			return true;
+		}
+		if self.cur_holder_commitment_transaction_number == INITIAL_COMMITMENT_NUMBER - 1 &&
+			self.cur_counterparty_commitment_transaction_number == INITIAL_COMMITMENT_NUMBER - 1 {
+			// If we're a 0-conf channel, and our commitment transaction numbers have both been
```

Comment:
Not sure I understand why `0-conf` is relevant here. Is the first paragraph of the comment describing how we would get inside the enclosing `if`? (Would prefer that the comment precede the `if` if that is the case.) And the second paragraph describing the assertion in the `if` immediately below?